src/components/TaskCard.tsx

```diff
@@ -0,0 +1,123 @@
+import React, { useState } from 'react';
+import { StyleSheet, Text, View, Image, TouchableOpacity } from 'react-native';
+import { SquircleView } from 'react-native-figma-squircle';
+import { ITask } from 'src/models/note';
+
+export interface ITaskCardProps {
+    note: ITask;
+}
+
+export const TaskCard: React.FC<ITaskCardProps> = props => {
+    const {
+        title,
+        text,
+        taskCardColor,
+        firsSubTaskColor,
+        secondSubTaskColor,
+        folder,
+        time,
+        cardBordersColor,
+        cardShadowColor,
+        isCompleted,
+    } = props.note;
+    const icon = {
+        emptyIcon: require('@assets/icons/not-marked-icon.png'),
+        notEmptyIcon: require('@assets/icons/marked-icon.png'),
+    };
+    const [isFinished, isUnfinished] = useState(isCompleted);
```

Comment:
Second element of an array returned by `useState` is not a variable. It is a setter function for first element. So, you should name it like
```suggestion
    const [isFinished, setIsFinished] = useState(isCompleted);
```
or even better if you wouldn't destruct variable `isCompleted` from props, so it wouldn't shadow inner variable
```suggestion
    const [isCompleted, setIsCompleted] = useState(task.isCompleted);
```
It is a good practice, because it is easier to read, when you jum from layout into component. You would see where exactly your prop is used. But if you rename it, it would a be a bit harder, as you'll need recognize what name your prop have inside component